docs/dark.md

```diff
@@ -0,0 +1,176 @@
+# DARK: Diophantine Argument of Knowledge
+
+**Reference**: [DARK paper](https://eprint.iacr.org/2003/105)  
+**Terms**:  [diophantine equation](../terms/diophantine_equation.md), [diophantine set](../terms/diophantine_set.md), [commitment
+scheme](../terms/commitment_scheme.md)
+
+![](attachments/dark.png)
+
+## 1. Integer Commitment Scheme
+
+Firstly, you should read the definition of [commitment scheme](../terms/commitment_scheme.md). DARK uses Damgard-Fujisaki integer
+commitment scheme, you can read the full paper of it
+[here](https://citeseerx.ist.psu.edu/document?repid=rep1&type=pdf&doi=fdfdb6df844a02ec1d783e43899a179920f13dac). To make explaining
+things much easier, this article will not explain the results of the paper in detail. There are two important properties of this
+commitment scheme you need to focus on:
+
+- **Summation Protocol** : If prover has three integer $a,b,c$ such that $a+b=c$ and their respective commitment $A,B,C$, they can
+  prove that $a+b=c$ in a zero-knowledge way using only the commitment values $A,B,C$.
+- **Multiplication Protocol** : If prover has three integer $a,b,c$ such that $a \cdot b=c$ and their respective commitment $A,B,C$,
+  they can prove that $a*b=c$ in a zero-knowledge way using only the commitment values $A,B,C$.
+
+In other words, the prover can claim that "these committed values actually have this sum/product relation" in a zero-knowledge way.
+Here is an easy example of how to prove that we have computed an integer coefficient multivariable polynomial on some input:
+
+- Prover has some values $x_1, x_2, x_3, x_4$ that satisfies $x_4 = x_1x_2+x_3$. How to prove it?
+- Prover calculate $x_5=x_1x_2$ and commit $x_1,x_2,x_3,x_4,x_5$.
+- Prover now can prove $x_5=x_1x_2$ and $x_4=x_3+x_5$ in a zero-knowledge proof, using only commitment.
+
+## 2. Diophantine Argument of Knowledge
+
+While integer commitment scheme is cool and already has quite a lot of applications, it's quite limited. There are plenty of
+prepositions that one would like to prove that does not have the form of "I evaluated a integer coefficient polynomial". Let's dig
+deeper.
+
+### 2.1 Diophantine Set
+
+You should read the definition of Diophantine set [here](../terms/diophantine_set.md).
+
+### 2.2 Arguments of Knowledge
+
+For bit-strings $a$ and $\mu$, and predicate $Q(·),$ we denote by $AK(Q(a,\mu))$ a three-round honest-verifier statistical
+zero-knowledge (==HVSZK==) two-party argument of knowledge (==AoK==) that given a value $a$ (known to both parties), the arguer knows
```

Comment:
i don't know the meaning of the symbol "==HVSZK==", or is it a math problem?

Reply:
This's a syntax for highlighting in Obsidian, but not in Github. It's a tradeoff.